File: superstitioApi/java/superstitioapi/hangUpCard/CardOrb_EachCardTrigger.java
package superstitioapi.hangUpCard;

import com.megacrit.cardcrawl.cards.AbstractCard;
import com.megacrit.cardcrawl.cards.CardGroup;
import com.megacrit.cardcrawl.orbs.AbstractOrb;
import org.apache.logging.log4j.util.BiConsumer;

public class CardOrb_EachCardTrigger extends CardOrb_CardTrigger implements ICardOrb_EachTime {
//    private int triggerTime;

    public CardOrb_EachCardTrigger(AbstractCard card, CardGroup cardGroupReturnAfterEvoke, int triggerTime, BiConsumer<CardOrb_CardTrigger,
            AbstractCard> action_thisOrb_triggerCard) {
        super(card, cardGroupReturnAfterEvoke, triggerTime, action_thisOrb_triggerCard);
    }

    @Override
    public AbstractOrb makeCopy() {
        return new CardOrb_EachCardTrigger(getOriginCard(), cardGroupReturnAfterEvoke, OrbCounter, action);
    }

    @Override
    protected boolean onProperCardUsed_checkIfShouldApplyAction(AbstractCard card) {
        return true;
    }
}

File: superstitioApi/java/superstitioapi/hangUpCard/CardOrb_EachCardTrigger.kt
package superstitioapi.hangUpCard

import com.megacrit.cardcrawl.cards.AbstractCard
import com.megacrit.cardcrawl.cards.CardGroup
import superstitioapi.utils.CostSmart

class CardOrb_EachCardTrigger  //    private int triggerTime;
    (
    card: AbstractCard,
    cardGroupReturnAfterEvoke: CardGroup?,
    triggerTime: CostSmart,
    action_thisOrb_triggerCard: (CardOrb_CardTrigger, AbstractCard) -> Unit
) : CardOrb_CardTrigger(card, cardGroupReturnAfterEvoke, triggerTime, action_thisOrb_triggerCard), ICardOrb_EachTime
{
    override fun makeCopy(): CardOrb
    {
        return CardOrb_EachCardTrigger(originCard, cardGroupReturnAfterEvoke, orbCounter, action)
    }

    override fun onProperCardUsed_checkIfShouldApplyAction(card: AbstractCard?): Boolean
    {
        return true
    }
}
